cmd/allocator/main.go

```diff
@@ -64,6 +63,18 @@ const (
 	sslPort = "8443"
 )
 
+// grpcHandlerFunc returns an http.Handler that delegates to grpcServer on incoming gRPC
+// connections or otherHandler otherwise. Copied from https://github.com/philips/grpc-gateway-example.
+func grpcHandlerFunc(grpcServer http.Handler, otherHandler http.Handler) http.Handler {
```

Comment:
Oh that's clever!

@roberthbailey we probably should have done this on the sdk sidecar! oh well!

Reply:
The other option is apparently to use a dedicated muxer but since Agones doesn't have one pulled in atm I figured it is easier to use this approach.

Reply:
Is this backwards compatible? I'm wondering if we could take the existing grpc port on the sidecar and add a similar handler to handle both grpc and http.

If so, we could migrate the default ports to be the same for both and then deprecate the ability to select a separate http port. After a grace period we could remove the ability to select the http port.

Reply:
My understanding is that it is backwards compatible in the sense that grpc clients will not be aware of the change when `grpcHandlerFunc` is added. I double checked it by running the example allocator-client.